src/CommandAndStateMessageParser.cpp

```diff
@@ -0,0 +1,397 @@
+
+    Json::FastWriter fast;
+    return fast.write(message);
+}
+
+Time CommandAndStateMessageParser::getTimeUsage(string address)
+{
+    double time = mJData["devices"][address]["currentSeconds"].asDouble();
+    return Time::fromSeconds(time);
+}
+
+RevolutionControl CommandAndStateMessageParser::getRevolutionControlStates(string address)
+{
+    auto msg_setpoint =
+        mJData["devices"][address]["control"]["setpoint"]["pose"]["localFrame"];
+    double setpoint_x = msg_setpoint["x"].asDouble();
+    double setpoint_y = msg_setpoint["y"].asDouble();
+    double setpoint_z = msg_setpoint["z"].asDouble();
+    double setpoint_yaw = msg_setpoint["yaw"].asDouble();
+
+    RevolutionControl control;
+    control.position.x() = setpoint_x;
+    control.position.y() = setpoint_y;
+    control.position.z() = setpoint_z;
+    control.orientation = Quaterniond(AngleAxisd(setpoint_yaw, Vector3d::UnitZ()));
+
+    return control;
+}
+
+RevolutionBodyStates CommandAndStateMessageParser::getRevolutionBodyStates(string address)
+{
+    auto local_frame =
+        mJData["devices"][address]["control"]["current"]["pose"]["localFrame"];
+    double state_x = local_frame["x"].asDouble();
+    double state_y = local_frame["y"].asDouble();
+    double state_z = local_frame["z"].asDouble();
+    double state_yaw = local_frame["yaw"].asDouble();
+
+    RevolutionBodyStates control;
+    control.position.x() = state_x;
+    control.position.y() = state_y;
+    control.position.z() = state_z;
+    control.orientation = Quaterniond(AngleAxisd(state_yaw, Vector3d::UnitZ()));
+
+    return control;
+}
+
+PoweredReelMotorStates CommandAndStateMessageParser::getPoweredReelMotorState(
+    string address)
+{
+    JointState state;
+    state.raw = mJData["devices"][address]["motor1Diagnostics"]["pwm"].asFloat() / 100;
+    state.effort = mJData["devices"][address]["motor1Diagnostics"]["current"].asDouble();
+    state.speed =
+        mJData["devices"][address]["motor1Diagnostics"]["rpm"].asFloat() * 2 * M_PI / 60;
+    PoweredReelMotorStates powered;
+    powered.elements.push_back(state);
+    state.raw = mJData["devices"][address]["motor2Diagnostics"]["pwm"].asFloat() / 100;
+    state.effort = mJData["devices"][address]["motor2Diagnostics"]["current"].asDouble();
+    state.speed =
+        mJData["devices"][address]["motor2Diagnostics"]["rpm"].asFloat() * 2 * M_PI / 60;
+    powered.elements.push_back(state);
+
+    return powered;
+}
+
+RevolutionMotorStates CommandAndStateMessageParser::getRevolutionMotorStates(
+    string address)
+{
+    JointState state;
+    state.raw =
+        mJData["devices"][address]["frontRightMotorDiagnostics"]["pwm"].asFloat() / 100;
+    state.effort =
+        mJData["devices"][address]["frontRightMotorDiagnostics"]["current"].asDouble();
+    state.speed =
+        mJData["devices"][address]["frontRightMotorDiagnostics"]["rpm"].asFloat() * 2 *
+        M_PI / 60;
```

Comment:
Create a `JoinState motorDiagnosticsToJointState(json)` helper function and use it here